Auth0/CredentialsManager.swift

```diff
@@ -0,0 +1,89 @@
+// CredentialsManager.swift
+//
+// Copyright (c) 2017 Auth0 (http://auth0.com)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import Foundation
+import SimpleKeychain
+
+/// Credentials management utility
+public struct CredentialsManager {
+
+    private let storage = A0SimpleKeychain()
+    private let storeKey = "credentials"
+    private let authentication: Authentication
+
+    /// Creates a new CredentialsManager instance
+    ///
+    /// - Parameters:
+    ///   - authentication: Auth0 authentication instance
+    public init(authentication: Authentication) {
+        self.authentication = authentication
+    }
+
+    /// Store credentials instance in keychain
+    ///
+    /// - Parameter credentials: credentials instance to store
+    /// - Returns: Bool outcome of success
+    public func store(_ credentials: Credentials) -> Bool {
```

Comment:
more for the flow of the api `manager.store(credentials: result)` then in the future we could have `manager.store(accessToken: token, expiresIn: date)` to avoid passing all the credential object.